pavelib/quality.py

```diff
@@ -315,6 +317,64 @@ def run_eslint(options):
         )
 
 
+def _get_stylelint_violations():
+    """
+    Returns the number of Stylelint violations.
+    """
+    stylelint_report_dir = (Env.REPORT_DIR / "stylelint")
+    stylelint_report = stylelint_report_dir / "stylelint.report"
+    _prepare_report_dir(stylelint_report_dir)
+    formatter = 'node_modules/stylelint-formatter-pretty'
+
+    sh(
+        "stylelint **/*.scss --custom-formatter={formatter} | tee {stylelint_report}".format(
+            formatter=formatter,
+            stylelint_report=stylelint_report,
+        ),
+        ignore_error=True
+    )
+
+    try:
+        return int(_get_count_from_last_line(stylelint_report, "eslint"))
+    except TypeError:
+        raise BuildFailure(
+            "Error. Number of eslint violations could not be found in {eslint_report}".format(
+                eslint_report=stylelint_report
+            )
+        )
+
+
+@task
+@needs('pavelib.prereqs.install_node_prereqs')
+@cmdopts([
+    ("limit=", "l", "limit for number of acceptable violations"),
+])
+@timed
+def run_stylelint(options):
+    """
+    Runs stylelint on Sass files.
+    If limit option is passed, fails build if more violations than the limit are found.
+    """
+    violations_limit = int(getattr(options, 'limit', -1))
+    num_violations = _get_stylelint_violations()
+
+    # Record the metric
+    _write_metric(num_violations, (Env.METRICS_DIR / "stylelint"))
+
+    # Fail if number of violations is greater than the limit
+    if num_violations > violations_limit > -1:
+        raise BuildFailure(
+            "Stylelint failed with too many violations: ({count}).\nThe limit is {violations_limit}.".format(
+                count=num_violations,
+                violations_limit=violations_limit,
+            )
+        )
+    if num_violations > 0:
+        print("Stylelint succeeded with no more violations than {violations_limit}".format(
```

Comment:
I don't think it makes sense to include `violations_limit`.  Did you mean to have something like?:
```
Stylelint succeeded with {num_violations} violations.
```

Reply:
I've just removed this too.